catalog/camel-route-parser/src/test/java/org/apache/camel/parser/helper/ParserCommonTest.java

```diff
@@ -0,0 +1,118 @@
+
+        JavaClassSource clazz = mock(JavaClassSource.class);
+        Block block = mock(Block.class);
+        SimpleName expression = mock(SimpleName.class);
+
+        try (MockedStatic<ParserCommon> parserCommonMockedStatic = mockStatic(ParserCommon.class)){
+            FieldSource<JavaClassSource> fieldSource = (FieldSource<JavaClassSource>) mock(FieldSource.class);
+            Type<JavaClassSource> type = (Type<JavaClassSource>) mock(Type.class);
+            when(fieldSource.getType()).thenReturn(type);
+            when(type.isType(typeParam)).thenReturn(true);
+            parserCommonMockedStatic.when(() -> ParserCommon.getField(clazz, block, expression)).thenReturn(fieldSource);
+            parserCommonMockedStatic.when(() -> ParserCommon.isNumericOperator(clazz, block, expression)).thenCallRealMethod();
+
+            assertTrue(ParserCommon.isNumericOperator(clazz, block, expression));
+
+            parserCommonMockedStatic.verify(() -> ParserCommon.getField(clazz, block, expression));
+            parserCommonMockedStatic.verify(() -> ParserCommon.isNumericOperator(clazz, block, expression));
+            parserCommonMockedStatic.verifyNoMoreInteractions();
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "byte", "Byte",        //      <- WARN probably a mistake?
+            "short", "Short",      //      <- WARN probably a mistake?
+            "double", "Double",    //      Floating Point Types
+            "float", "Float",      //      Floating Point Types
+            "char", "Character", "String",
+            "boolean", "Boolean"})
+    @SuppressWarnings("unchecked")
+    void isNumericOperatorShouldReturnFalseWhenExpressionInstanceofSimpleNameTest(String typeParam) {
+        JavaClassSource clazz = mock(JavaClassSource.class);
+        Block block = mock(Block.class);
+        SimpleName expression = mock(SimpleName.class);
+
+        try (MockedStatic<ParserCommon> parserCommonMockedStatic = mockStatic(ParserCommon.class)){
+            FieldSource<JavaClassSource> fieldSource = (FieldSource<JavaClassSource>) mock(FieldSource.class);
+            Type<JavaClassSource> type = (Type<JavaClassSource>) mock(Type.class);
+            when(fieldSource.getType()).thenReturn(type);
+            when(type.isType(typeParam)).thenReturn(true);
+            parserCommonMockedStatic.when(() -> ParserCommon.getField(clazz, block, expression)).thenReturn(fieldSource);
+            parserCommonMockedStatic.when(() -> ParserCommon.isNumericOperator(clazz, block, expression)).thenCallRealMethod();
+
+            assertFalse(ParserCommon.isNumericOperator(clazz, block, expression));
+
+            parserCommonMockedStatic.verify(() -> ParserCommon.getField(clazz, block, expression));
+            parserCommonMockedStatic.verify(() -> ParserCommon.isNumericOperator(clazz, block, expression));
+            parserCommonMockedStatic.verifyNoMoreInteractions();
+        }
+    }
```

Comment:
I think I would probably move these 2 tests to a separate file. Then, I'd separate the test setup from the verification part and would use the `@DisplayName` annotation to make the expectations clearer.